packages/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -130,7 +146,7 @@ public function get_declarations_string( $should_prettify = false, $indent_count
 
 		foreach ( $declarations_array as $property => $value ) {
 			$spacer               = $should_prettify ? ' ' : '';
-			$filtered_declaration = esc_html( safecss_filter_attr( "{$property}:{$spacer}{$value}" ) );
+			$filtered_declaration = esc_html( static::filter_declaration( $property, $value, $spacer ) );
```

Comment:
For a follow up: could we abstract all sanitization and move the `esc_html` too?

Reply:
Good question — depending on the output context, I wasn't too sure if `esc_html()` is always desired so left it inline for the time being. Happy to play around with it in a follow-up if this PR looks viable (I'd like to better understand when and why we need to escape the html characters, too — I can imagine some circumstances, but I don't feel I fully get it yet 🙂).

Reply:
I think this is worth investigating separately. The `safecss_filter_attr` is probably enough here.

Looking at other usages of `safecss_filter_attr` in the plugin, I don't see any of them wrapped in the `esc_*` method.

Reply:
> could we abstract all sanitization and move the esc_html too?

`esc_html` causes issues with values that contain quotes - like for example URLs in `background:url("example.com/image.jpg")`. That's why over in https://github.com/WordPress/gutenberg/pull/42968/files I was introducing a `sanitize_value` method 😉

Reply:
I did some tests, and now I'm 99% sure we don't need `esc_html` here.

Sorry for the ping, @peterwilsoncc. Can you confirm that values returned by `safecss_filter_attr` don't need escaping?

```php
$value = safecss_filter_attr( $css );

// VS

$value = esc_html( safecss_filter_attr( $css ) );
```

Reply:
I think we can just remove `esc_html`
It's not used anywhere in Core when printing CSS, and the safecss call takes care of most (all?) cases I can think of that could potentially cause issues.

Reply:
Thanks for the discussion, folks! I've updated this PR to remove the call to `esc_html` and updated the tests accordingly.